### PR TITLE
refactor: introduce basic list-item

### PR DESCRIPTION
**Pull request by kelsos**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by kelsos** [comment]
Maybe we can adjust this and use it to migrate some of the cases at least

**Comment by kelsos** [comment]
@lukicenturi if this needs anything pick it up and do any changes needed

File: frontend/app/src/components/common/ListItem.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      'px-4': !noPadding,
+      'hover:bg-rui-grey-100 hover:dark:bg-rui-grey-800': !noHover
+    }"
+    v-on="
+      // eslint-disable-next-line vue/no-deprecated-dollar-listeners-api
+      $listeners
+    "
+  >
+    <div
+      v-if="$slots.avatar"
```
Comment (tewshi) [inline frontend/app/src/components/common/ListItem.vue:44]: thought we always prefer to use useSlots

Comment (kelsos) [reply frontend/app/src/components/common/ListItem.vue:44]: So this in mostly because of a misunderstanding I had on what gets removed.

Comment (lukicenturi) [reply frontend/app/src/components/common/ListItem.vue:44]: So I should change it to use `useSlots` right? @kelsos

Comment (kelsos) [reply frontend/app/src/components/common/ListItem.vue:44]: Anything is fine we can figure out what makes sense after we finish the upgrade

**Review by tewshi — APPROVED** [review]
lgtm!

**Pull request merged**